src/Microsoft.DocAsCode.Build.Common/YamlHtmlPart.cs

```diff
@@ -0,0 +1,60 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace Microsoft.DocAsCode.Build.Common
+{
+    using System.Collections.Generic;
+
+    using HtmlAgilityPack;
+
+    public class YamlHtmlPart
+    {
+        public HtmlDocument Doc { get; set; }
+        public int StartLine { get; set; }
+        public int EndLine { get; set; }
+
+        public static IList<YamlHtmlPart> SplitYamlHtml(string html)
+        {
+            var doc = new HtmlDocument();
+            doc.LoadHtml(html);
+            var parts = new List<YamlHtmlPart>();
+
+            var nodes = doc.DocumentNode.SelectNodes("//yamlheader");
+            if (nodes == null)
+            {
+                return parts;
+            }
+
+            foreach (var node in nodes)
+            {
+                var part = new HtmlDocument();
+
+                var startLineStr = node.GetAttributeValue("start", "-1");
+                var endLineStr = node.GetAttributeValue("end", "-1");
```

Comment:
is it possible to use public int GetAttributeValue(string name, int def);? [http://docs.workflowgen.com/wfgmy/v240/html/586e2308-155c-c399-0a44-6662610e5ae5.htm](http://docs.workflowgen.com/wfgmy/v240/html/586e2308-155c-c399-0a44-6662610e5ae5.htm)